### PR TITLE
Remove TL_CRON usage from newsletter-bundle

### PR DESCRIPTION
**Pull request by fritzmg**

### CODE REVIEW

File: newsletter-bundle/src/Cron/PurgeSubscriptionsCron.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        }
+
+        /** @var NewsletterRecipientsModel $subscription */
+        foreach ($subscriptions as $subscription) {
+            $subscription->delete();
+        }
+
+        if (null !== $this->logger) {
+            $this->logger->info('Purged the unactivated newsletter subscriptions');
+        }
```
Comment (ausi) [inline newsletter-bundle/src/Cron/PurgeSubscriptionsCron.php]:
```suggestion
        $this->logger?->info('Purged the unactivated newsletter subscriptions');
```

File: newsletter-bundle/src/Resources/contao/classes/Newsletter.php [hunk 2]
```diff
@@ -932,9 +932,13 @@ public function updateAccount()
 
 	/**
 	 * Purge subscriptions that have not been activated within 24 hours
+	 *
+	 * @deprecated Deprecated since Contao 5.0, to be removed in Contao 6.0.
 	 */
 	public function purgeSubscriptions()
 	{
+		trigger_deprecation('contao/newsletter-bundle', '5.0', 'Calling "%s::%s()" has been deprecated and will no longer work in Contao 6.0.', __CLASS__, __METHOD__);
```
Comment (ausi) [inline newsletter-bundle/src/Resources/contao/classes/Newsletter.php]:
```suggestion
		trigger_deprecation('contao/newsletter-bundle', '5.0', 'Calling "%s()" has been deprecated and will no longer work in Contao 6.0.', __METHOD__);
```

Comment (fritzmg) [reply newsletter-bundle/src/Resources/contao/classes/Newsletter.php]: Good catch, there are actually a few places in the core where `__CLASS__` and `__METHOD__` is used in the deprecation message like this.

**Review by m-vo — DISMISSED** [review]
LGTM besides the two annotations by @ausi.

File: newsletter-bundle/src/Resources/contao/classes/Newsletter.php (hunk 2 above)
Comment (ausi) [reply newsletter-bundle/src/Resources/contao/classes/Newsletter.php]: Yes, I used it like that in the past I think 🙈

Comment (fritzmg) [reply newsletter-bundle/src/Resources/contao/classes/Newsletter.php]: https://github.com/contao/contao/pull/4643 🙃

**Review by ausi — DISMISSED** [review]

**Review by m-vo — DISMISSED** [review]

**Review by Toflar — DISMISSED** [review]

**Review by ausi — DISMISSED** [review]

File: newsletter-bundle/src/Cron/PurgeSubscriptionsCron.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+{
+    public function __construct(private ContaoFramework $framework, private LoggerInterface|null $logger)
+    {
+    }
+
+    public function __invoke(): void
+    {
+        $this->framework->initialize();
+
+        $subscriptions = $this->framework->getAdapter(NewsletterRecipientsModel::class)->findExpiredSubscriptions();
```
Comment (aschempp) [inline newsletter-bundle/src/Cron/PurgeSubscriptionsCron.php:33]: I was confused by this variable name. They are not subscriptions but recipients, maybe you want to change that 🙃

Comment (fritzmg) [reply newsletter-bundle/src/Cron/PurgeSubscriptionsCron.php:33]: I used that variable name because of the method name.

**Review by leofeyer — CHANGES_REQUESTED** [review]
Same here. Changes are OK but the unit test is never run unless you set them up.

**Comment by fritzmg** [comment]
> Same here. Changes are OK but the unit test is never run unless you set them up.

Not sure what you mean, the newsletter-bundle already has tests and it's already registered in the `phpunit.xml.dist`?

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @fritzmg.

**Pull request merged**